antelope_contracts/contracts/erc20/src/erc20.cpp

```diff
@@ -345,10 +345,17 @@ void erc20::setfee(eosio::name token_contract, eosio::symbol token_symbol, const
 
     token_table_t token_table(_self, _self.value);
     auto index_symbol = token_table.get_index<"by.symbol"_n>();
-    auto iter = index_symbol.find(v);
-    check(iter != index_symbol.end(), "token not registered");
+    auto token_table_iter = index_symbol.find(v);
+    eosio::check(token_table_iter != index_symbol.end(), "token not registered");
 
+    message_receiver_table message_receivers(evm_account, evm_account.value);
+    auto message_receivers_iter = message_receivers.find(erc2o_account.value);
+    eosio::check(message_receivers_iter != message_receivers.end(), "receiver not registered in evm contract");
+
+    intx::uint256 min_fee((uint64_t)message_receivers_iter->min_fee.amount);
     intx::uint256 egress_fee_evm = egress_fee.amount;
+    eosio::check(egress_fee_evm >= min_fee, "egress fee to be set must be larger than minimum fee");
```

Comment:
This makes it confusing in terms of units.

Since we have restricted the interface of this action to only take an `egress_fee` with a precision of what `eosio::asset` supports (which technically there is no reason we necessarily need to do it that way by the way), then we should just simply compare `eosio::asset` to `eosio::asset`.

Just check the following:
```cpp
eosio::check(egress_fee >= message_receivers_iter->min_fee, "egress fee must be at least as large as the receiver's minimum fee");
```

Notice I changed the error message too. Your existing message implies it has to be strictly greater which is not true.

Reply:
changed